### PR TITLE
feat(provider): clicksend sms provider addition

### LINKED ISSUE #4388: [NV-2968] 🚀 Feature: Add Clicksend SMS Provider
### 🔖 Feature description

Use the [Clicksend Docs](https://developers.clicksend.com/docs/rest/v3/?javascript--nodejs#ClickSend-v3-API-SMS) to create a new SMS Provider while following [Novu's guide to new providers](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider)

### 🎤 Why is this feature needed ?

add providers for the community

### ✌️ How do you aim to achieve this?

community contributions

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-2968](https://linear.app/novu/issue/NV-2968/🚀-feature-add-clicksend-sms-provider)</sub>

### PR DESCRIPTION
**Pull request by parveshsaini**
### What change does this PR introduce?

This PR introduces Clicksend SMS provider.

Doc Reference: https://developers.clicksend.com/docs/rest/v3/?javascript--nodejs#send-sms

### Why was this change needed?

- Closes https://github.com/novuhq/novu/issues/4388

### Other information (Screenshots)

WhatsApp Image 2023-10-10 at 09 06 30_954b77a7

WhatsApp Image 2023-10-10 at 09 10 07_79e165c7

WhatsApp Image 2023-10-10 at 09 07 02_d886c8fe

WhatsApp Image 2023-10-10 at 09 08 13_dc94af72

### CODE REVIEW

File: providers/clicksend/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+## Usage
+
+```javascript
+    import { ClicksendSmsProvider } from '@novu/clicksend'
+
+    const provider = new ClicksendSmsProvider({
+        username: process.env.CLICKSEND_USERNAME,
+        apiKey: process.env.CLICKSEND_API_KEY,
+```
```
Comment (BiswaViraj) [inline providers/clicksend/README.md:13]:
```suggestion
    const provider = new ClicksendSmsProvider({
        username: process.env.CLICKSEND_USERNAME,
        apiKey: process.env.CLICKSEND_API_KEY,
        })
    ```
```

Comment (BiswaViraj) [reply providers/clicksend/README.md:13]: Closing brackets are missing

**Review by BiswaViraj — CHANGES_REQUESTED** [review]

File: providers/clicksend/README.md (hunk 1 above)
Comment (parveshsaini) [reply providers/clicksend/README.md]: Done 👍

**Comment by parveshsaini** [comment]
Changes done

File: packages/application-generic/package.json [hunk 2]
```diff
@@ -111,7 +111,8 @@
     "pino-pretty": "^9.4.0",
     "redlock": "4.2.0",
     "reflect-metadata": "^0.1.13",
-    "rrule": "^2.7.2"
+    "rrule": "^2.7.2",
+    "@novu/clicksend": "^0.16.3"
```
Comment (BiswaViraj) [inline packages/application-generic/package.json:115]:
```suggestion
    "@novu/clicksend": "0.20.0-alpha.1"
```

File: providers/clicksend/package.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
+{
+  "name": "@novu/clicksend",
+  "version": "0.16.3",
```
Comment (BiswaViraj) [inline providers/clicksend/package.json:3]:
```suggestion
  "version": "0.20.0-alpha.1",
```

**Review by BiswaViraj — CHANGES_REQUESTED** [review]
Thank you for your contribution🌟.
Please update the version of the package. I will merge it

**Comment by parveshsaini** [comment]
Version updated. Thankyou!

**Review by BiswaViraj — APPROVED** [review]
Awesome 🚀

File: providers/clicksend/src/lib/clicksend.provider.spec.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  describe('sendMessage', () => {
+    it('should return a response with an id and date', async () => {
+      const response = await provider.sendMessage({
+        to: '+0451111111',
+        content: 'test message',
+      });
+
+      expect(response).toHaveProperty('id');
```
Comment (BiswaViraj) [inline providers/clicksend/src/lib/clicksend.provider.spec.ts:24]: You'll need to mock the response to make the tests pass.
For reference check `clickatell` providers spec file, how the sendMessage method is mocked.

**Review by BiswaViraj — CHANGES_REQUESTED** [review]

File: providers/clicksend/src/lib/clicksend.provider.spec.ts (hunk 4 above)
Comment (parveshsaini) [reply providers/clicksend/src/lib/clicksend.provider.spec.ts]: Hey @BiswaViraj! I've updated the spec file by mocking the response. The tests passed as well.
Please let me know if there's anything else required from my side :)

WhatsApp Image 2023-10-11 at 19 43 50_6e90890d

Comment (BiswaViraj) [reply providers/clicksend/src/lib/clicksend.provider.spec.ts]: Thank you, merging it now

**Review by BiswaViraj — APPROVED** [review]

**Pull request merged**